src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2912,33 +2912,30 @@ def _ensure_service_principal(cli_ctx,
     # TODO: This really needs to be unit tested.
     rbac_client = get_graph_rbac_management_client(cli_ctx)
     if not service_principal:
-        # --service-principal not specified, try to load it from local disk
-        principal_obj = load_acs_service_principal(subscription_id)
-        if principal_obj:
-            service_principal = principal_obj.get('service_principal')
-            client_secret = principal_obj.get('client_secret')
-        else:
-            # Nothing to load, make one.
-            if not client_secret:
-                client_secret = _create_client_secret()
-            salt = binascii.b2a_hex(os.urandom(3)).decode('utf-8')
-            url = 'https://{}.{}.{}.cloudapp.azure.com'.format(salt, dns_name_prefix, location)
-
-            service_principal = _build_service_principal(rbac_client, cli_ctx, name, url, client_secret)
-            if not service_principal:
-                raise CLIError('Could not create a service principal with the right permissions. '
-                               'Are you an Owner on this project?')
-            logger.info('Created a service principal: %s', service_principal)
-            # add role first before save it
-            if not _add_role_assignment(cli_ctx, 'Contributor', service_principal):
-                logger.warning('Could not create a service principal with the right permissions. '
-                               'Are you an Owner on this project?')
+        # --service-principal not specified, make one.
```

Comment:
A small suggestion here is that can we try to utilize [existing function](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/role/custom.py#L1268)

Reply:
I am a bit hesitant to make this change. `create_service_principal_for_rbac` in the `role` module only takes `name` (and derives `url` from it for creating the service principal), whereas the present function takes `name` and `dns_name_prefix`. I see three options:

1. Reuse `create_service_principal_for_rbac` in the `role` module **as-is**: This would lead to a change in the way the URL or name of SPs created by `az aks create` are generated.
2. Reuse `create_service_principal_for_rbac`, but change its definition so it takes a URL.
3. Keep the present solution, i.e., do not reuse `create_service_principal_for_rbac`.

Let me know what you prefer.

Reply:
Got it. I am fine to all three options. 😊 Then please take a look for CI error.

Reply:
Fixed, thanks for reviewing!